rq/worker.py

```diff
@@ -1160,3 +1164,22 @@ def request_force_stop_sigrtmin(self, signum, frame):
         info = dict((attr, getattr(frame, attr)) for attr in self.frame_properties)
         self.log.warning('raising ShutDownImminentException to cancel job...')
         raise ShutDownImminentException('shut down imminent (signal: %s)' % signal_name(signum), info)
+
+
+class RoundRobinWorker(Worker):
+    """
+    Modified version of Worker that dequeues jobs from the queues using a round-robin strategy.
+    """
+    def reorder_queues(self, reference_queue):
+        pos = self.queues.index(reference_queue)
+        self.queues=self.queues[pos+1:]+self.queues[:pos+1]
```

Comment:
I don't think we should mutate `self.queues`. Can you create another property `self._ordered_queues` that we use for this purpose?

Reply:
No problem about adding a new property, but I would like to know where it should be used (apart from the call to `dequeue_any` in line 650). For instance, I see no point in using it in `validate_queues` (line 241), but I am not sure if it should be used in `queue_names` (line 247) or `queue_keys` (line 251). Same for the call to `RQScheduler` in line 553 and the definitions of `refresh`(line 729) and `clean_registries` (line 1088).

Reply:
`_ordered_queues` is used to shuffle queue orderings. `self.queues` shouldn't be mutated.